ibm-messaging-mq-cloud-showcase-app/backend/AWS-showcase-backend/msms/message-session-manager.js

```diff
@@ -90,6 +90,89 @@ class MQClient {
     this.myID = uuidv4();    
   }
 
+  poisonMessageHandler(){
+    debug_info("Inside the poison message handler");
+    let cno = new mq.MQCNO();
+    cno.Options = MQC.MQCNO_CLIENT_BINDING;
+    let csp = new mq.MQCSP();
+    csp.UserId = credentials.USER;
+    csp.Password = credentials.APP_PASSWORD;
+    cno.SecurityParms = csp;
+
+    let cd = new mq.MQCD();
+    cd.ChannelName = MQDetails.CHANNEL;
+    cd.ConnectionName = this.getConnection();
+    cno.ClientConn = cd;
+
+    mq.ConnxSync("QM1", cno, function (err, hConn) {
```

Comment:
Why is there a fresh connection and a fresh get? Don't we already have the message? A commit would delete it off the queue.

As it stands this poison message handling is only invoked when a get has failed. The subsequent get in this function would in that case also fail. 

This poison message handling should be handling the case where a message has been obtained, but it can't be processed. eg. because it doesn't have a valid reply to queue. In which case the message has been read. A subsequent read, will not get the same message. 

If running as a transaction, the transaction either needs to be rolled back, to put the message back on the queue or committed to take it off.

If not running inside a transaction, then the message is already off the queue, another read (in this poison message handler) will read the next message.

Have these changes been tested?

Reply:
> Why is there a fresh connection and a fresh get? Don't we already have the message? A commit would delete it off the queue.

The way I tested this was using the `basicrequest` sample of the Node.js samples. So when the message is put on the queue, and I switched on the responder, the backend got a http `500` error, and the message used to go back right into the queue. That's why the logic which I applied for this was to open the queue, get the message and manually put it into the poison queue, since otherwise it would be in the main queue the entire time.